storage/src/archive/storage.rs

```diff
@@ -76,9 +65,8 @@ impl<T: Translator, K: Array, B: Blob, E: Storage<B> + Metrics> Archive<T, K, B,
     ) -> Result<Self, Error> {
         // Initialize keys and run corruption check
         let mut indices = BTreeMap::new();
-        let mut keys = HashMap::new();
+        let mut keys = Index::init(context.with_label("archive"), cfg.translator.clone());
```

Comment:
Let's use `index` here (the context provided to `archive` will likely already have some archive-related label).